### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by hpractv**
Created an action that takes a zip path, copies over to a folder on a target IIS server, and unzips the files.

### CODE REVIEW

**Review by JosephDSchwartz — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,75 +1,93 @@
...
+      ...
 
-      - name: Add the action here
-        uses: im-open/this-repo@v1.0.0
+      deploy-package:
+      uses: 'im-open/deploy-on-prem-web-app@v1.0.0'
         with:
-          input-1: 'abc'
-          input-2: '123
+          server: ${{ env.server }}
```
Comment (danielle-casella-adams) [inline README.md:85]: You don't have to but it might be nice to add the server up in your env list above.

File: action.yml [hunk 2]
```diff
@@ -1,22 +1,45 @@
...
-    value: '${{ steps.step_1.outputs.world }}'
 
 runs:
   using: 'composite'
   steps:
-    - name: Step 1
-      id: step_1
-      shell: bash
+    - name: Action
+      id: action
```
Comment (danielle-casella-adams) [inline action.yml]: You don't have to give this a name or id since it's so generic.

File: README.md [hunk 3]
```diff
@@ -1,75 +1,93 @@
...
-    | microsoft-teams | For actions related to Microsoft Teams   |
-    | svc             | For actions related to Windows Services  |
-    | jira            | For actions related to Jira              |
-    | meta            | For actions related to running workflows |
-    | pagerduty       | For actions related to PagerDuty         |
-    | test            | For actions related to testing           |
-    | tf              | For actions related to Terraform         |
-  - [ ] Add any additional topics for an action if they apply    
-    
+# Deploy On-Premises Web App
```
Comment (danielle-casella-adams) [inline README.md:1]: It  might be nice to add somewhere that this is only going to work on a windows runner.

**Review by danielle-casella-adams — APPROVED** [review]
I have some comments that I wouldn't hold the PR up for if you don't want to address them, but take a look and see if you want to address any of them.

File: README.md (hunk 1 above)
Comment (hpractv) [reply README.md:85]: 👍🏽

File: action.yml (hunk 2 above)
Comment (hpractv) [reply action.yml]: 👍🏽

File: README.md [hunk 4]
```diff
@@ -1,79 +1,105 @@
...
+| `service-account-id`       | true        | The service account name                                                          |
+| `service-account-password` | true        | The service account password                                                      |
+| `source-zip-file-path`     | true        | The path to the zip file that contains the application files                      |
+| `deployment-folder-path`   | true        | The path to deploy the application files to                                       |
+| `clean-deployment-folder`  | false       | Remove files that are not in the source zip file, accepts values of true or false |
+| `server-public-key`        | true        | Path to remote server public ssl key (local path like c:\folder)                  |
+
+## Prerequisites
+
+The On-Premises Web Application Deployment action uses Web Services for Management, [WSMan], and Windows Remote Management, [WinRM], to create remote administrative sessions. Because of this Self-Hosted, Windows OS GitHubs Actions Runners, `runs-on: [self-hosted, windows-2019]`, must be used.  This will also allow the actions runner access to our on-premises applications servers.
```
Comment (danielle-casella-adams) [inline README.md]: If someone outside of our org stumbles across this they're going to read that self-hosted is required.  Would that be true for them?  Is this only going to work on our own customized action runners?

If so I might put another disclaimer saying this action has been customized for im-open's use and may not work outside of our organization.

Comment (hpractv) [reply README.md]: I'll try to clarify in a generic way.

**Pull request merged**